<launch>
  <!-- default values for all critical parameters, can be set from cmdline when running the launch file -->
  <arg name="takeoff_height" default="1.50"/>
  <arg name="constant_yaw" default="true" />
  <arg name="wp_radius" default="0.25"/>
  <arg name="wp_mode" default="0"/>
  <arg name="exec_mode" default="0"/>
  <arg name="en_voxl_offset" default="false"/>
  <arg name="en_avoid" default="false"/>
  <arg name="en_dynamic_avoid" default="false"/>
  <arg name="en_prec_land" default="false"/>
  <arg name="en_vio_fs" default="false"/>
  <arg name="waypoint_hold_time" default="0.0"/>
  <arg name="tag_data_src" default="0"/>
  <arg name="hdg_offset_deg" default="0.0"/>
  <arg name="read_extern_yaw" default="false"/>
  <arg name="align_yaw_before_move" default="true"/>
  <arg name="enable_wall_following" default="false"/>

  <node pkg="rsun_autonomy" type="autonomy_manager.py" name="waypoint_tracker" output="screen" cwd="node">
    <!-- Params -->
    <rosparam command="load" file="$(find rsun_autonomy)/cfg/config_params.yaml"/>

    <!-- User configured values overwrite values in the config_param.yaml file -->
    <param name="takeoff_height" value="$(arg takeoff_height)"/>    
    <param name="wp_mode" value="$(arg wp_mode)"/>
    <param name="exec_mode" value="$(arg exec_mode)"/>
    <param name="yaw_hold" value="$(arg constant_yaw)"/>
    <param name="wp_radius" value="$(arg wp_radius)"/>
    <param name="waypoint_hold_time" value="$(arg waypoint_hold_time)"/>  
    <param name="enable_voxl_offset" value="$(arg en_voxl_offset)"/>
    <param name="enable_avoidance" value="$(arg en_avoid)"/>
    <param name="enable_dynamic_avoidance" value="$(arg en_dynamic_avoid)"/>
    <param name="enable_prec_land" value="$(arg en_prec_land)"/>
    <param name="enable_vio_failsafe" value="$(arg en_vio_fs)"/>
    <param name="fiducial_tag_type" value="$(arg tag_data_src)"/>
    <param name="heading_offset_deg" value="$(arg hdg_offset_deg)"/>
    <param name="read_extern_yaw" value="$(arg read_extern_yaw)"/>
    <param name="align_yaw_before_move" value="$(arg align_yaw_before_move)"/>
    <param name="enable_wall_following" value="$(arg enable_wall_following)"/>
    
    <!-- Main Topics out -->
    <remap from="/waypoint_markers" to="/waypoint_markers"/>

  </node>

</launch>